### PR TITLE
Improve automatic MinorStep when MajorStep is 2 (#133)

### PR DESCRIPTION
**Pull request by Jonarw**
I noticed this problem when I created some plots. I found this old issue and thought I'd commit my solution. I hope it is ok not to update the changelog for such a minor change?

### CODE REVIEW

**Comment by Jonarw** [comment]
Ahm, I didn't know new commits are automatically added to the pull request... The request however is only meant for the first commit ;)

**Comment by objorke** [comment]
Great!! I am really looking forward to merge this :-)
Minor change? With 400 lines of math? I think this is a nice enhancement that should be mentioned in the change log.
The broken build seems to be an appveyor nuget restore issue. I try rebuilding the PR.
I will add some more comments in the code.

File: Source/OxyPlot/Axes/Axis.cs [hunk 1]
```diff
@@ -1685,22 +1685,13 @@ protected double CalculateActualInterval(double availableSize, double maxInterva
         /// <returns>The minor interval.</returns>
         protected double CalculateMinorInterval(double majorInterval)
```
Comment (objorke) [inline Source/OxyPlot/Axes/Axis.cs]: This method should be covered by unit tests, I think. The new implementation is not totally clear to me.. :-)
I would try test cases: 1, 2, 5, 10, 25 etc.

File: Source/OxyPlot/Axes/LogarithmicAxis.cs [hunk 2]
```diff
@@ -78,80 +78,211 @@ public override void CoerceActualMaxMin()
...
+            double axisBandwidth = this.IsVertical() ? this.ScreenMax.Y - this.ScreenMin.Y : this.ScreenMax.X - this.ScreenMin.X;
+            double logAxisBandwidth = Math.Log(axisBandwidth, this.Base);
+
+            double desiredNumberOfTicks = axisBandwidth / this.IntervalLength;
+            double ticksPerDecade = desiredNumberOfTicks / logBandwidth;
+            double desiredStepSize = 1.0 / Convert.ToInt32(ticksPerDecade);
+
+            int intBase = Convert.ToInt32(this.Base);
+            
+            if (ticksPerDecade < 0.75)
```
Comment (objorke) [inline Source/OxyPlot/Axes/LogarithmicAxis.cs]: the cyclomatic complexity of this method has increased *a lot*. Can smaller methods be extracted? https://sourcemaking.com/refactoring/smells/long-method

**Comment by objorke** [comment]
We need at least one example in the example library to show how this works!

**Comment by objorke** [comment]
Yes, new commits to the same branch are automatically added to the PR.
You should create a new branch for each PR.

**Comment by objorke** [comment]
The build is ok now :)

**Comment by objorke** [comment]
I think this PR should be closed, and a new one created from your branch minorinterval2

**Pull request closed without merging**